### PR TITLE
Implement CPU plugin just-in-time emitter for Mod operation

### PR DESCRIPTION
**Pull request by awayzjj**
Closes https://github.com/openvinotoolkit/openvino/issues/24061
### Details:
 - Mod operation

### Tickets:
 - [CVS-137689](https://jira.devtools.intel.com/browse/CVS-137689)

### CODE REVIEW

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp [hunk 1]
```diff
@@ -391,6 +391,47 @@ std::set<std::vector<element::Type>> jit_exp_emitter::get_supported_precisions(c
...
+
+    using TReg = typename dnnl::impl::cpu::aarch64::cpu_isa_traits<isa>::TReg;
+
+    TReg divend = TReg(in_vec_idxs[0]);
+    TReg divisor = TReg(in_vec_idxs[1]);
+    TReg r = TReg(out_vec_idxs[0]);
+
+    h->uni_fdiv(r.s, divend.s, divisor.s);
+    h->uni_fmul(r.s, r.s, divisor.s);
+    h->uni_fsub(r.s, divend.s, r.s);
```
Comment (a-sidorova) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:428]: Regarding your question in https://github.com/openvinotoolkit/openvino/issues/24061#issuecomment-2118860639:
This emitter works with FP32 values. It means that `uni_fmul` instruction is inverse for the previous instruction `uni_fdiv`.
For example,
```
divend = 10.f
divisor = 3.f
h->uni_fdiv(r.s, divend.s, divisor.s); // r.s = 10.f / 3.f = 3.(33)f
h->uni_fmul(r.s, r.s, divisor.s); // r.s = 3.(33)f * 3.f ~= 10.f
h->uni_fsub(r.s, divend.s, r.s); // r.s = 10.f - 10.f = 0
```
So you need to truncate fractional part of division result before `fmul` call.

Comment (awayzjj) [reply src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp]: @a-sidorova  Thank you very much for your detailed suggestions. I have incorporated your suggestions and passed the tests. Could you please review the PR?

**Comment by eshoguli** [comment]
@awayzjj, thanks for contribution. Looks like you are close to finishing. Please, rebase onto the latest remote master.

**Comment by ilya-lavrenov** [comment]
build_jenkins

**Comment by awayzjj** [comment]
@eshoguli I have merged the master branch master into my branch.

But three CI checks(
`ie_tests_cpu_spr_ubuntu20_release`, `ie_tests_cpu_debian_9_arm_release`, `build_win_vs2019_release`) failed.

Could you please show me some logs of failed test cases or give me some suggestions? Thank you very much!

**Comment by ilya-lavrenov** [comment]
build_jenkins

**Comment by awayzjj** [comment]
@eshoguli @a-sidorova Hi, 2 CI checks(`ie_tests_cpu_debian_9_arm_release`, `ci/jenkins`) failed, but I could not see any logs.

Could you please show me some logs of failed test cases or give me some suggestions? Thank you very much!

**Comment by a-sidorova** [comment]
> @eshoguli @a-sidorova Hi, 2 CI checks(`ie_tests_cpu_debian_9_arm_release`, `ci/jenkins`) failed, but I could not see any logs.
> 
> Could you please show me some logs of failed test cases or give me some suggestions? Thank you very much!

There are 38 failed tests `smoke_CompareWithRefs_(4|5)D_MemOrder.*eltwise_op_type=Mod_.*` with the same exception:
> ../../../../../../../repos/openvino/src/plugins/intel_cpu/tests/functional/utils/cpu_test_utils.cpp:220
Value of: primTypeCheck(primType)
  Actual: false
Expected: true
primType is unexpected : ref_f32 Expected : acl_f32

**Comment by dmitry-gorokhov** [comment]
@awayzjj Have you tried to run related tests locally? The issue description contains details what tests should be run to check the implementation. I expect that running these tests on ARM machine will show the same error CI is showing.
image

**Comment by awayzjj** [comment]
@a-sidorova @dmitry-gorokhov Thank you for your reply! I failed to reproduce the failed test cases on my ARM server with ubuntu18.04 as the following results shows.
```
(base) root@ecs-7714:/mnt/data/openvino# cat /proc/version
Linux version 4.15.0-70-generic (buildd@bos02-arm64-021) (gcc version 7.4.0 (Ubuntu/Linaro 7.4.0-1ubuntu1~18.04.1)) #79-Ubuntu SMP Tue Nov 12 10:36:10 UTC 2019
```

```
./bin/aarch64/Debug/ov_cpu_func_tests --gtest_filter="*smoke_CompareWithRefs_4D_MemOrder*eltwise_op_type=Mod_*" --gtest_break_on_failure
```

截屏2024-05-22 22 15 23
截屏2024-05-22 22 15 35

```
./bin/aarch64/Debug/ov_cpu_func_tests --gtest_filter="*smoke_CompareWithRefs_5D_MemOrder*eltwise_op_type=Mod_*" --gtest_break_on_failure
```

截屏2024-05-22 22 23 18
截屏2024-05-22 22 23 25

```
nohup ./bin/aarch64/Debug/ov_cpu_func_tests --gtest_filter="*smoke*Eltwise*" &
```

截屏2024-05-22 22 28 18

 Do I have to switch to debian 9 since the `ie_tests_cpu_debian_9_arm_release` job failed but  `ie_tests_cpu_ubuntu20_arm64_release` job succeed?
But it is hard for me to find such a ARM machine with debian 9 :)

**Comment by awayzjj** [comment]
@dmitry-gorokhov Thank you very much!  I have incorporated your suggestions. Could you please review the PR?

**Comment by dmitry-gorokhov** [comment]
build_jenkins

**Comment by awayzjj** [comment]
@dmitry-gorokhov The `ie_tests_cpu_debian_9_arm_release` job passed!

But the `ie_tests_ie_python_api_arm_ubuntu20_arm64_release` failed, could you please show me the failed test cases?

**Comment by dmitry-gorokhov** [comment]
> @dmitry-gorokhov The `ie_tests_cpu_debian_9_arm_release` job passed!
> 
> But the `ie_tests_ie_python_api_arm_ubuntu20_arm64_release` failed, could you please show me the failed test cases?

The fails seem to be not related to your changes. I restarted the jobs to check CI instability.

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
@awayzjj The PR is merged. Thanks for the contribution!

**Pull request merged**